src/lib/features/feature-toggle/configuration-revision-service.ts

```diff
@@ -0,0 +1,48 @@
+import { EventEmitter } from 'stream';
+import { Logger } from '../../logger';
+import { IEventStore, IUnleashConfig, IUnleashStores } from '../../types';
+
+export const UPDATE_REVISION = 'UPDATE_REVISION';
+
+export default class ConfigurationRevisionService extends EventEmitter {
+    private logger: Logger;
+
+    private eventStore: IEventStore;
+
+    private revisionId: number;
+
+    constructor(
+        { eventStore }: Pick<IUnleashStores, 'eventStore'>,
+        { getLogger }: Pick<IUnleashConfig, 'getLogger'>,
+    ) {
+        super();
+        this.logger = getLogger(
+            'features/config-client/toggle-config-change-emitter',
```

Comment:
Nitpicking: I think we're usually doing this:
```suggestion
            'configuration-revision-service.ts',
```
But maybe there's benefit in having the full path (although I think there's a copy&paste leftover here):
```suggestion
            'features/feature-toggle/configuration-revision-service',
```

Reply:
GH double-posted this? :-1: